lib/tool_shed/tools/tool_validator.py

```diff
@@ -291,16 +291,10 @@ def load_tool_from_config(self, repository_id, full_path):
             tool = create_tool_from_source(config_file=full_path, app=self.app, tool_source=tool_source, repository_id=repository_id, allow_code_files=False)
             valid = True
             error_message = None
-        except KeyError as e:
-            tool = None
-            valid = False
-            error_message = 'This file requires an entry for "%s" in the tool_data_table_conf.xml file.  Upload a file ' % str(e)
-            error_message += 'named tool_data_table_conf.xml.sample to the repository that includes the required entry to correct '
-            error_message += 'this error.  '
-        except Exception as e:
-            tool = None
-            valid = False
-            error_message = str(e)
+        except Exception:
+            # As best I can tell, error messages will no longer get back to the UI, so raise exceptions instead so that
```

Comment:
That doesn't look right to me, there's some additional logic in the calling code that checks if a tool is valid and, if it makes sense, displays `error_message`. Otherwise no need to even return `valid` and `error_message`, I think.

Reply:
So the situation I was encountering was an exception due to a change to the method signature of `Tool.__init__()`. The clone was succeeding, the tool was "installed" as far as the install DB was concerned, but it wasn't being added to `shed_tool_conf.xml`. There was no error reported in the UI or log file. So although the error was of my own making, figuring it out was a real nightmare. Presumably there are other non-development errors that would be missed by this as well?

Reply:
I'm sure there are, but this breaks all code that uses the current logic. How about just logging the exception here?